pkg/odo/cli/component/deploy.go

```diff
@@ -98,8 +98,16 @@ func (do *DeployOptions) Validate() (err error) {
 	}
 
 	s = log.Spinner("Validating build information")
+
 	metadata := do.devObj.Data.GetMetadata()
```

Comment:
We should move this line `metadata := do.devObj.Data.GetMetaData()` done where `metadata` is used.